fem/lininteg.cpp

```diff
@@ -497,22 +500,24 @@ void BoundaryFlowIntegrator::AssembleRHSElementVect(
       Tr.Loc1.Transform(ip, eip);
       el.CalcShape(eip, shape);
 
-      Tr.Face->SetIntPoint(&ip);
+      Tr.SetIntPoint(&ip);
+      Tr.SetActiveSide(0);
 
-      u->Eval(vu, *Tr.Elem1, eip);
+      // Use Tr transformation in case u or f depends on boundary attribute
+      u->Eval(vu, Tr, ip);
```

Comment:
This is also incorrect. The coefficients `u` and `f` are "volume" coefficients -- they are the same as the ones used for `DGTraceIntegrator` and the volume term, `ConvectionIntegrator`, uses the product of the two.

Replacing `*Tr.Elem1` with `Tr` does not work simply because of `Tr.SetActiveSide(0)`. It looks like `SetActiveSide` is misleading and on top of that it is not really necessary. It is only used with 0 and boundary faces. Also, it should not be used for interior faces (inside integrator implementations) because most coefficient do not support this and there is no need to support it.

Reply:
When you say these are "volume" coefficients do you mean that they should be determined by a volume attribute rather than a boundary attribute?  If so, this should be documented someplace because it is counter intuitive.

I'm not sure what you mean about `SetActiveSide` being misleading. Is this related to the ambiguity about using volume or boundary attributes?

Reply:
Hi, @v-dobrev , the only use of this integrator that I can find is in `ex9` and `ex9p`.  From looking at these it makes sense that `u` should be a "volume" coefficient.  However, `f` is not used in the other integrators appearing in example 9.  Also, the name used for `f` in that example is `inflow` which strongly suggests that it should be associated with a boundary.  Are you sure we want `f` to be a "volume" coefficient?  I could add an optional argument to the constructor to indicate which transformation object should be used with `f` if that flexibility would be useful.

Reply:
You are right, `f` is the inflow boundary condition which makes sense to be evaluated as a "boundary" coefficient. I got confused with the weight "volume" coefficient, `rho`, which can be used in the `DGTraceIntegrator`.